### PR TITLE
Added demo folder

### PR DESCRIPTION
**Pull request by asgrim**
Just a folder with code that should work with current version of the library that demonstrates usage in an isolated way

### CODE REVIEW

File: demo/basic-reflection/index.php [hunk 1]
```diff
@@ -0,0 +1,46 @@
+<?php
+
+// Load a standard (internal) class
```
Comment (Ocramius) [inline demo/basic-reflection/index.php]: These should be separate files IMO

Comment (asgrim) [inline demo/basic-reflection/index.php]: done, yo

Comment (Ocramius) [inline demo/basic-reflection/index.php]: @asgrim merged, yo

**Comment by asgrim** [comment]
thanks

**Pull request merged**